protocol/contracts/amo/AuraStaking.sol

```diff
@@ -0,0 +1,142 @@
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/AMO__IBaseRewardPool.sol";
+import "./interfaces/AMO__IAuraBooster.sol";
+import "./helpers/AMOCommon.sol";
+
+contract AuraStaking is Ownable {
+    using SafeERC20 for IERC20;
+
+    address public operator;
+    // @notice BPT tokens for balancer pool
+    IERC20 public bptToken;
+    AuraPoolInfo public auraPoolInfo;
+    // @notice Aura booster
+    AMO__IAuraBooster public immutable booster;
+    IERC20 public immutable depositToken;
+
+    struct AuraPoolInfo {
+        address token;
+        address rewards;
+        uint32 pId;
+    }
+
+    struct Position {
+        uint256 staked;
+        uint256 earned;
+    }
+
+    error NotOperator();
+    error NotOperatorOrOwner();
+
+    event SetAuraPoolInfo(uint32, address, address);
+    event SetOperator(address);
+    event RecoveredToken(address, address, uint256);
+
+    constructor(
+        address _operator,
+        IERC20 _bptToken,
+        AMO__IAuraBooster _booster,
+        IERC20 _depositToken
+    ) {
+        operator = _operator;
+        bptToken = _bptToken;
+        booster = _booster;
+        depositToken = _depositToken;
+    }
+
+    function setAuraPoolInfo(uint32 _pId, address _token, address _rewards) external onlyOwner {
+        auraPoolInfo.pId = _pId;
+        auraPoolInfo.token = _token;
+        auraPoolInfo.rewards = _rewards;
+
+        emit SetAuraPoolInfo(_pId, _token, _rewards);
+    }
+
+    /**
+     * @notice Set operator
+     * @param _operator New operator
+     */
+    function setOperator(address _operator) external onlyOwner {
+        operator = _operator;
+
+        emit SetOperator(_operator);
+    }
+
+    /**
+     * @notice Recover any token from AMO
+     * @param token Token to recover
+     * @param to Recipient address
+     * @param amount Amount to recover
+     */
+    function recoverToken(address token, address to, uint256 amount) external onlyOwner {
+        if (to == address(0)) revert AMOCommon.InvalidAddress();
+        IERC20(token).safeTransfer(to, amount);
+
+        emit RecoveredToken(token, to, amount);
+    }
+
+    function depositAndStake(uint256 amount) external onlyOperator {
+        bptToken.safeTransferFrom(msg.sender, address(this), amount);
+
+        bptToken.safeIncreaseAllowance(address(booster), amount);
+        booster.deposit(auraPoolInfo.pId, amount, true);
+    }
+
+    // withdraw deposit token and unwrap to bpt tokens
+    function withdrawAndUnwrap(uint256 amount, bool claim, bool sendToOperator) external onlyOperatorOrOwner {
+        IBaseRewardPool(auraPoolInfo.rewards).withdrawAndUnwrap(amount, claim);
+        if (sendToOperator) {
+            // unwrapped amount is 1 to 1
+            bptToken.safeTransfer(operator, amount);
+        }
+    }
+
+    function withdrawAllAndUnwrap(bool claim, bool sendToOperator) external onlyOwner {
```

Comment:
Should this be onlyOperatorOrOwner too, to be consistent with withdrawAndUnwrap?

Reply:
`onlyOwner`. it's a fallback for multisig to withdraw all bpt tokens. operator just deposits and withdraws calculated bPT amounts